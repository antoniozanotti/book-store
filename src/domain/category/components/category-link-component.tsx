import { CategoryLinkComponentInterface } from "./category-link-component-interface";
import { getCategoryUrl } from "../helpers/get-category-url";
import Link from "next/link";

export function CategoryLinkComponent({
  category,
  children,
  ...props
}: CategoryLinkComponentInterface) {
  const link = getCategoryUrl(category);
  
  return (
    <Link href={link} title={category.name} className={props.className}>
      {children}
    </Link>
  );
}
