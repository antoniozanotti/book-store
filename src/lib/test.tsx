import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
