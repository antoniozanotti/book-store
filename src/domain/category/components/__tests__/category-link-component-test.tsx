import { render, screen } from "@/lib/test";
import { CategoryLinkComponent } from "../category-link-component";
import { categoriesWithPageMock } from "../../__tests__/categories-with-page-mock";

describe("CategoryLinkComponent", () => {
  it("should render", async () => {
    const { container } = render(
      <CategoryLinkComponent category={categoriesWithPageMock[0]}>
        Test
      </CategoryLinkComponent>
    );
    expect(container).toMatchSnapshot();
  });
});
