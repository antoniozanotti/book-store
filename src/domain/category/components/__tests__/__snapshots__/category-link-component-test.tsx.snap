// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryLinkComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <a
      href="/category/arts-and-entertainment"
      title="Arts and Entertainment"
    >
      Test
    </a>
  </div>
</div>
`;
